src/BenchmarksApps/Kestrel/PlatformBenchmarks/BufferExtensions.cs

```diff
@@ -26,34 +26,32 @@ internal static unsafe void WriteNumeric<T>(ref this BufferWriter<T> buffer, uin
             const byte AsciiDigitStart = (byte)'0';
 
             var span = buffer.Span;
-            var bytesLeftInBlock = span.Length;
 
             // Fast path, try copying to the available memory directly
             var advanceBy = 0;
-            fixed (byte* output = span)
+            if (span.Length >= 3)
```

Comment:
there is no need to pin this span, with size hint we always execute the "happy path" with size big enough and no boundaries check when indexing